src/handlers/user-identity-handler.ts

```diff
@@ -1,33 +1,97 @@
+import { decodeJwt, JWTPayload } from "jose";
+
 import logger from "../commons/logger";
-import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
-import { HTTP_MESSAGE_INVALID_REQUEST, HttpCodesEnum } from "../types/constants";
-
-export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
-  logger.debug("Received message", { event: event.body });
-  if (!event.body) {
-    return {
-      statusCode: HttpCodesEnum.BAD_REQUEST,
-      body: JSON.stringify({ message: HTTP_MESSAGE_INVALID_REQUEST }),
-    };
-  }
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from "aws-lambda";
+import { HttpCodesEnum } from "../types/constants";
+import { getConfiguration, getServiceApiKey } from "../types/configuration";
+import { EvcsStoredIdentityResponse, StoredIdentityResponse, UserIdentityDataType } from "../types/interfaces";
+
+interface ErrorResponse {
+  error: string;
+  error_description: string;
+}
+
+export const handler = async (event: APIGatewayProxyEvent, context: Context): Promise<APIGatewayProxyResult> => {
+  logger.addContext(context);
+
   if (!event?.headers?.Authorization) {
-    return {
-      statusCode: HttpCodesEnum.UNAUTHORIZED,
-      body: JSON.stringify({ message: "Request Unauthorized" }),
-    };
+    logger.error("Authorisation header was not included in request");
```

Comment:
With all of the logging, we should probably be including the govuk-journey-id (it's in the request body) but maybe this is a separate story that needs writing up. Don't want to inflate this story further.